service/src/test/java/com/epam/esm/service/impl/CertificateServiceTest.java

```diff
@@ -0,0 +1,139 @@
+import java.math.BigInteger;
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestConfig.class)
+class CertificateServiceTest {
+
+    @Autowired
+    private CertificateService certificateService;
+
+    @Autowired
+    private CertificateDAO certificateDAO;
+
+    @Autowired
+    private CertificateConverter certificateConverter;
+
+    @Autowired
+    @Qualifier("tagServiceMock")
+    private TagService tagService;
+
+    @Autowired
+    private CertificateValidator certificateValidator;
+
+    private final static String FIRST_NAME = "Apple.com";
+    private final static BigInteger FIRST_ID = BigInteger.ONE;
+    private final static BigInteger SECOND_ID = BigInteger.TWO;
+    private final static String SECOND_NAME = "OZ.com";
+    private final static String DESCRIPTION = "Certificate description";
+    private final static BigDecimal PRICE = BigDecimal.TEN;
+    private final static int DURATION = 42;
+    private final static String TAG_NAME = "#cool";
+    private final static String SORT_BY_NAME = "ASC";
+    private final static LocalDateTime CREATE_DATE = LocalDateTime.now();
+    private final static LocalDateTime LAST_UPDATE_DATE = LocalDateTime.now();
+    private final static int DAO_UPDATE_RETURN_VALUE = 3;
+    private final Certificate firstCertificate =
+            new Certificate(FIRST_ID, FIRST_NAME, DESCRIPTION, PRICE, DURATION, CREATE_DATE, LAST_UPDATE_DATE);
+    private final Certificate secondCertificate =
+            new Certificate(SECOND_ID, SECOND_NAME, DESCRIPTION, PRICE, DURATION, CREATE_DATE, LAST_UPDATE_DATE);
+
+
+    @Test
+    public void findAllTest() {
+        final List<Certificate> certificates = List.of(firstCertificate, secondCertificate);
+        when(certificateConverter.convert(any(Certificate.class))).thenReturn(new CertificateDTO());
+        when(certificateDAO.findAll()).thenReturn(certificates);
+        final Collection<CertificateDTO> certificateDTOS = certificateService.findAll();
+        assertNotNull(certificateDTOS);
+        assertFalse(certificateDTOS.isEmpty());
+        final long count = certificateDTOS.size();
+        assertEquals(count, 2);
+    }
+
+    @Test
+    public void findAllByTagIdTest() {
+        final List<Certificate> certificates = List.of(firstCertificate, secondCertificate);
+        when(certificateConverter.convert(any(Certificate.class))).thenReturn(new CertificateDTO());
+        when(certificateDAO.findAll(any(BigInteger.class))).thenReturn(certificates);
+        final Collection<CertificateDTO> certificateDTOS = certificateService.findAll(FIRST_ID);
+        assertNotNull(certificateDTOS);
+        assertFalse(certificateDTOS.isEmpty());
+        final int count = certificateDTOS.size();
+        assertEquals(count, 2);
+    }
+
+    @Test
+    public void findAllWithTagsByParamsTest() {
```

Comment:
It would be a great idea to use given-when-then comments to increase the readability of your tests